### DISCUSSION TITLE
Qualquer consulta na API, pelo proprio swagger, tenta acessar antes /Account/Login

### DISCUSSION THREAD

**Discussion by henriquebg88** [balta-io/forum#719] [Tira Dúvidas] [answered]
Imagino que tenha alguma coisa errada na minha configuração de Endpoints para o Identity, ou configuração de autorização.. 
Comparei com os arquivos do curso e não conseguir identificar diferença.. 
![image](https://github.com/user-attachments/assets/5a154d41-283d-4c81-8b70-b9f4d986929e)

### **Endpoints.cs**
`
using Dima.API.Commom.API;
using Dima.API.Endpoints.Categories;
using Dima.API.Endpoints.Identity;
using Dima.API.Models;

namespace Dima.API.Endpoints
{
    public static class Endpoint
    {
        private const string V1 = "v1";
        private const string CATEGORIES = "categories";
        private const string TRANSACTIONS = "transactions";
        private const string IDENTITY = "identity";

        public static void MapEndpoints(this WebApplication app)
        {
            var endpointsV1 = app.MapGroup("");
            
            app.MapGroup("")
                .WithTags("Health check")
                .MapGet("/", () => new {message = "Estou vivo."});

            endpointsV1.MapGroup($"{V1}/{CATEGORIES}")
                       .WithTags(CATEGORIES)
                       .RequireAuthorization()
                       .MapEndpoint<CreateCategoryEndpoint>()
                       .MapEndpoint<UpdateCategoryEndpoint>()
                       .MapEndpoint<DeleteCategoryEndpoint>()
                       .MapEndpoint<GetByIdCategoryEndpoint>()
                       .MapEndpoint<GetAllCategoryEndpoint>();

            endpointsV1.MapGroup($"{V1}/{TRANSACTIONS}")
                       .WithTags(TRANSACTIONS)
                       .RequireAuthorization()
                       .MapEndpoint<CreateTransactionEndpoint>()
                       .MapEndpoint<UpdateTransactionEndpoint>()
                       .MapEndpoint<DeleteTransactionEndpoint>()
                       .MapEndpoint<GetByIdTransactionEndpoint>()
                       .MapEndpoint<GetTransactionsByPeriodEndpoint>();

            endpointsV1.MapGroup($"{V1}/{IDENTITY}")
                       .WithTags(IDENTITY)
                       .MapIdentityApi<User>();

            endpointsV1.MapGroup($"{V1}/{IDENTITY}")
                       .WithTags(IDENTITY)
                       .MapEndpoint<LogoutEndpoint>()
                       .MapEndpoint<GetRolesEndpoint>();
        }

        private static IEndpointRouteBuilder MapEndpoint<TEndpoint>(this IEndpointRouteBuilder endpoint) where TEndpoint : IEndpoint
        {
            TEndpoint.Map(endpoint);
            return endpoint;
        }
        
    }
}
`
### **Program.cs**
`
using System.Security.Claims;
using Dima.API;
using Dima.API.Commom.API;
using Dima.API.Endpoints;
using Dima.API.Models;
using Microsoft.AspNetCore.Identity;

var builder = WebApplication.CreateBuilder(args);

builder.AddDimaConfiguration();
builder.AddSecurity();
builder.AddDataContexts();
builder.AddDimaCors();
builder.AddSwaggerDocumentation();
builder.AddDimaServices();                

var app = builder.Build();

if (app.Environment.IsDevelopment()){
    app.ConfigureDevEnviroment();
}

app.UseCors(APIConfiguration.CorsPolicyName);

app.UseSecurity();

app.MapEndpoints();

app.Run();
`

### **BuilderExtensions.cs**
`
using Dima.API.Data;
using Dima.API.Handlers;
using Dima.API.Models;
using Dima.Core;
using Dima.Core.Handlers;
using Microsoft.AspNetCore.Identity;
using Microsoft.EntityFrameworkCore;

namespace Dima.API.Commom.API
{
    public static class BuilderExtension
    {
        public static void AddDimaConfiguration(this WebApplicationBuilder builder){
            Configurations.connectionString = builder.Configuration.GetConnectionString("Default") ?? "";
            Configurations.FrontendURL = builder.Configuration.GetValue<string>("FrontendURL") ?? "";
            Configurations.BackendURL = builder.Configuration.GetValue<string>("BackendURL") ?? "";
        }

        public static void AddSwaggerDocumentation(this WebApplicationBuilder builder){
            builder.Services.AddEndpointsApiExplorer();
            builder.Services.AddSwaggerGen( SW => {
                SW.CustomSchemaIds( n => n.FullName );
            });
        }

        public static void AddSecurity(this WebApplicationBuilder builder)
        {
            builder.Services.AddAuthentication(IdentityConstants.ApplicationScheme)  
                .AddIdentityCookies();
            builder.Services.AddAuthorization(); 
        }

        public static void AddDataContexts(this WebApplicationBuilder builder){
            builder.Services.AddDbContext<AppDbContext>( x => {
                            x.UseSqlServer(Configurations.connectionString);
            });

            builder.Services.AddIdentityCore<User>()
                            .AddRoles<IdentityRole<long>>()
                            .AddEntityFrameworkStores<AppDbContext>(
                            .AddApiEndpoints(); 
   
        }

        public static void AddDimaServices(this WebApplicationBuilder builder){
           
            builder.Services.AddTransient<ICaterogyHandler, CategoryHandler>(); 
            builder.Services.AddTransient<ITransactionHandler, TransactionHandler>(); 
        }

        public static void AddDimaCors(this WebApplicationBuilder builder){
            builder.Services.AddCors(options => {
                options.AddPolicy(
                    APIConfiguration.CorsPolicyName,
                    policy => policy
                        .WithOrigins([
                            Configurations.BackendURL,
                            Configurations.FrontendURL
                        ])
                        .AllowAnyMethod()
                        .AllowAnyHeader()
                        .AllowCredentials()
                );
            });
        }
    }
}
`

### **AppExtension.cs**

`
using System.Security.Claims;
using Dima.API.Models;
using Microsoft.AspNetCore.Identity;

namespace Dima.API.Commom.API
{
    public static class AppExtension
    {
        public static void ConfigureDevEnviroment(this WebApplication app){
            app.UseSwagger();
            app.UseSwaggerUI();
            app.MapSwagger().RequireAuthorization();
        }

        public static void UseSecurity(this WebApplication app){
            app.UseAuthentication();
            app.UseAuthorization();
        }
        
    }
}
`

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @henriquebg88, este "erro" ocorre por que você está tentando acessar um endpoint que precisa estar logado sem estar logado... Por isso recebe essa mensagem de erro!

**Reply by henriquebg88** [reply]
Não estou conseguindo fazer o login.. 

![image](https://github.com/user-attachments/assets/acb94683-a61f-4a9a-8186-cab9c934ea65)

**Reply by andrebaltieri** [reply]
Bom dia, @henriquebg88 você selecionou a opção "UseCookies", ao que me parece o erro que está dando é por que ele não encontrou um esquema para autenticação via Token (Que é a padrão), mas no curso utilizamos via Cookie!
